### DISCUSSION TITLE
Pressure calculation in MultiphasePoromechanics

### DISCUSSION THREAD

**Discussion by alexandrebtessael** [GEOS-DEV/GEOS#2020] [Q&A]
Hello. I performed several simulations using _MultiphasePoromechanics_ solver and I observed some irregularities regarding the calculation of the pressure in some areas of my meshes (see screenshot below). I expected the pressure to increase in the reservoir because of gas injection but some cells showed a decrease in pressure instead, which is a phenomenon that I only observe while using _MultiphasePoromechanics_, not with _CompositionalMultiphaseFlow_ alone. Also, these cells with pressure drops (or lower pressures) are intertwined with cells where the pressure increases which seems odd.

![image](https://user-images.githubusercontent.com/87317386/185422119-e39ef106-0f21-46dd-a181-dac234109b97.png)

I managed to reproduce this type of behaviour on a cube 5x5x5 where I compared the injection of gas using a SourceTerm in a reservoir filled with water. I did two simulations using the same parameters for rock properties, initial and boundary conditions.

My first simulation uses the _CompositionalMultiphaseFlow_ solver. The pressure increases everywhere in the cube after each time-step which is what I expected.
The second simulation is performed with the _MultiphasePoromechanics_ solver and as you can see in the screenshot below, some areas stay blue and show a decrease in pressure in some parts of the cube.

![image](https://user-images.githubusercontent.com/87317386/185414861-b2d4ea2a-26ed-47de-b063-ae23dd65469c.png)

I put these 2 simulations file here : https://filesender.renater.fr/?s=download&token=8ca5fcbe-bd10-42b1-9f5e-469002407db5
Is this behaviour normal for this solver ? How to prevent it ?

**Comment by jhuang2601** [comment]
In the testing case with cube, pore pressure perturbation is too small if compared to the initial pore pressure. Can you try to increase injection rate and duration to avoid this data oscillation for the poromechanical case?

**Comment by paveltomin** [comment]
we observed similar behavior on a field-scale case
can confirm that it is related to mechanical effects and does not occur for flow-only simulation
CC @herve-gross
